### PR TITLE
[docs] make `bar-code-scanner.md` more accessible & correct

### PR DESCRIPTION
**Pull request by MaeWolff**
# Why

- Removing the use of IIFE as it is hard for many newcomers to wrap their heads around.

# How

:)

# Test Plan

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/versions/unversioned/sdk/bar-code-scanner.md [hunk 1]
```diff
@@ -67,10 +67,12 @@ export default function App() {
   const [scanned, setScanned] = useState(false);
 
   useEffect(() => {
-    (async () => {
+    const getBarCodeScannerPermissions = async () => {
       const { status } = await BarCodeScanner.requestPermissionsAsync();
-      setHasPermission(status === 'granted');
-    })();
+      setHasPermission(status === 'granted' ? true : false);
```
Comment (Simek) [inline docs/pages/versions/unversioned/sdk/bar-code-scanner.md:72]:
```suggestion
      setHasPermission(status === 'granted');
```

Hello @MaeWolff, thank you for adding name to the permission effect function for the clarity. 👍 

Before we merge it I have a question: What's the reasoning behind updating the line above? Compare like that [can only yield a Boolean result](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_equality).

Comment (MaeWolff) [reply docs/pages/versions/unversioned/sdk/bar-code-scanner.md:72]: Hi @Simek ! 

There's not really reasoning, I thought it was more readable at first sight. I can remove my modification if you want

Comment (Simek) [reply docs/pages/versions/unversioned/sdk/bar-code-scanner.md:72]: Let's revert to the previous form than, and we can ship it. 🙂

Comment (MaeWolff) [reply docs/pages/versions/unversioned/sdk/bar-code-scanner.md]: [4562eaa](https://github.com/expo/expo/pull/17747/commits/4562eaacdf29f765a8003f8f39f99411c01bf21e) :)

**Review by Simek — APPROVED** [review]
LGTM, thank you for the update! 👍

**Pull request merged**